### ISSUE TITLE
Setting accessibilityLabel to testID in production is not a good idea

### ISSUE THREAD

**Issue by LeoNatan** [isnifer/react-native-testid#1]
`accessibilityLabel` is used by the iOS system to help handicapped users. For instance, Voice Over uses this.

If you set some testing value to this, your handicapped users will suffer.

**Comment by LeoNatan** [comment]
Differentiate [accessibilityIdentifier](https://developer.apple.com/documentation/uikit/uiaccessibilityidentification/1623132-accessibilityidentifier?language=objc) and [accessibilityLabel](https://developer.apple.com/documentation/objectivec/nsobject/1615181-accessibilitylabel?language=objc).

**Comment by isnifer** [comment]
@LeoNatan look into the code. I'm using `testID` for iOS.

**Comment by LeoNatan** [comment]
On Android it’s exactly the same problem:
https://developer.android.com/training/accessibility/accessible-app.html

**Comment by LeoNatan** [comment]
Google TalkBack uses that to read the content description to the user. Exactly the same as Voice Over on iOS.

**Comment by isnifer** [comment]
@LeoNatan sadly, but Appium does not work otherwise

**Comment by LeoNatan** [comment]
Then fix Appium. Disabled users should not suffer because some test framework is buggy. Sorry

**Comment by isnifer** [comment]
@LeoNatan yes, of course

**Comment by isnifer** [comment]
@LeoNatan we are using Appium until Detox doesn't work with it like with iOS :)

**Comment by LeoNatan** [comment]
It is being worked on actively.

**Comment by simonracz** [comment]
Just a side note: Detox Android works with test id already.

Appium can't see test ids, because they are transferred to view tags on Android. (You probably know about this.)
View tags are not accessible for UIAutomator, on which Appium depends on.

**Comment by isnifer** [comment]
@simonracz yes, I know. Thank you. 
If I understand correctly I need only `testID` for both platforms if I will use `detox` only, yes?

**Comment by LeoNatan** [comment]
Yes, that is the best way to test. `byText` is not optimal because if you localization, it will fail your test.

**Comment by isnifer** [comment]
@LeoNatan good to know. So, for now, we are using Appium only to check Auth via Twitter and FB, also to test Apple Pay, cause EarlGrey can't do that.
